app/controllers/users.js

```diff
@@ -87,9 +89,21 @@ const findAll = (request, response) =>
 
 const findUserAlbums = (request, response) =>
   users
-    .findById(request.params.user_id)
+    .findById(request.params.id)
     .then(user => user.getAlbums())
     .then(albums => response.send(albums))
     .catch(_logDatabaseError(response));
 
-module.exports = { save, login, findAll, saveOrUpdateAdmin, findUserAlbums };
+const findUserAlbumsPhotos = (request, response) =>
+  users
+    .findById(request.params.id)
+    .then(user => user.getAlbums())
+    .then(albums =>
+      Promise.all(
+        Array.prototype.map.call(albums, album => axios.get(`${photoEndpoint}?albumId=${album.id}`))
```

Comment:
albums.map?

Reply:
Sadly I can't, albums is not an iterable, it's an object that contains some iterable functions (like `forEach` and `filter`), but when I tried to use `map` directly, it was `undefined`.

Reply:
oh, got it